### PR TITLE
Ensure THIRD-PARTY-LICENCES.txt is checked in

### PR DESCRIPTION
**Pull request by GFJHogue**
Main issue: #3763

Added a step in CircleCI to verify that generated `THIRD-PARTY-LICENSES.txt` matches the checked-in version.
Also removed first-party `io.dockstore` licenses from third-party license file.

### CODE REVIEW

File: scripts/check-licenses.sh [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+set -o nounset
+set -o xtrace
+
+if [ "${TESTING_PROFILE}" != "automated-review" ]; then
+    exit 0;
+fi
+
+cp THIRD-PARTY-LICENSES.txt generated-licenses.txt
+git checkout THIRD-PARTY-LICENSES.txt
+diff generated-licenses.txt THIRD-PARTY-LICENSES.txt
```
Comment (coverbeck) [inline scripts/check-licenses.sh:16]: No newline, per GitHub

File: scripts/check-licenses.sh [hunk 2]
```diff
@@ -0,0 +1,16 @@
+#!/usr/bin/env bash
+# Checks that the generated THIRD-PARTY-LICENSES.txt matches the checked-in file
+# Includes Bash3 Boilerplate. Copyright (c) 2014, kvz.io
+
+set -o errexit
+set -o pipefail
+set -o nounset
+set -o xtrace
+
+if [ "${TESTING_PROFILE}" != "automated-review" ]; then
```
Comment (coverbeck) [inline scripts/check-licenses.sh:10]: I'm not clear why we need this, although it is following the convention of propose_migration. 🤷

**Review by coverbeck — APPROVED** [review]

File: scripts/check-licenses.sh (hunk 2 above)
Comment (GFJHogue) [reply scripts/check-licenses.sh:10]: Me neither. The other similar scripts seem to do it ¯\\\_(ツ)\_/¯

Comment (denis-yuen) [reply scripts/check-licenses.sh:10]:
```
      - run:
          name: check that JPA classes are consistent with migrations
          command: |
            scripts/check_migrations.sh
          environment:
            TESTING_PROFILE: automated-review
```
It seems to run one in circle. But check_migrations uses `if [ "${TESTING_PROFILE}" = "automated-review" ]; then` to only run in that phase. This seems to be the opposite, so it always runs?

Would have thought this should also run only in one phase (and edit the description linked)

**Review by denis-yuen — COMMENTED** [review]
Comment, not sure if phase running is intentional

File: scripts/check-licenses.sh (hunk 2 above)
Comment (GFJHogue) [reply scripts/check-licenses.sh:10]: They both behave the same way.

The main code for `check_migrations` is inside of the if block.
The main code that I have for `check-licenses` is outside of the if block (easier to inherit the exit code of `diff` when not skipping).

Comment (denis-yuen) [reply scripts/check-licenses.sh:10]: > They both behave the same way.

Ah, cool. Missed that.

Then just update the name above to include that `check that JPA classes are consistent with migrations and that third party licenses are up-to-date`

**Review by denis-yuen — APPROVED** [review]
~With minor comment~

**Review by garyluu — APPROVED** [review]

**Review by NatalieEO — APPROVED** [review]

**Pull request merged**